pilot/pkg/proxy/envoy/v2/ads_common_test.go

```diff
@@ -25,24 +25,90 @@ import (
 )
 
 func TestProxyNeedsPush(t *testing.T) {
+	const (
+		invalidKind = "INVALID_KIND"
+		svcName     = "svc1.com"
+		drName      = "dr1"
+		vsName      = "vs1"
+	)
+
+	pushResourceScopeBak := pushResourceScope
+	pushResourceScope = map[resource.GroupVersionKind]func(*model.Proxy, *XdsEvent, map[string]struct{}) bool{
+		model.ServiceEntryKind: func(proxy *model.Proxy, pushEv *XdsEvent, resources map[string]struct{}) bool {
+			_, _ = proxy, pushEv
+			if len(resources) == 0 {
+				return true
+			}
+			_, f := resources[svcName]
+			return f
+		},
+		model.VirtualServiceKind: func(proxy *model.Proxy, pushEv *XdsEvent, resources map[string]struct{}) bool {
+			_, _ = proxy, pushEv
+			if len(resources) == 0 {
+				return true
+			}
+			_, f := resources[vsName]
+			return f
+		},
+		model.DestinationRuleKind: func(proxy *model.Proxy, pushEv *XdsEvent, resources map[string]struct{}) bool {
+			_, _ = proxy, pushEv
+			if len(resources) == 0 {
+				return true
+			}
+			_, f := resources[drName]
+			return f
+		},
+	}
+	defer func() {
+		pushResourceScope = pushResourceScopeBak
```

Comment:
Is it feasible to refactor this a bit to avoid modifying global state?

Reply:
Ok, I'll try that.